### PR TITLE
chore: render env stack updates from the env controller

### PR DESCRIPTION
**Pull request by efekarakus**
https://user-images.githubusercontent.com/879348/106506161-2edfae00-647e-11eb-9aee-42d1e5d81b1e.mov

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: internal/pkg/stream/cloudformation.go [hunk 1]
```diff
@@ -124,9 +124,7 @@ func (s *StackStreamer) Notify() {
 	// notifying previous subscribers of older events.
 	s.mu.Lock()
 	var subs []chan StackEvent
-	for _, sub := range s.subscribers {
-		subs = append(subs, sub)
-	}
+	subs = append(subs, s.subscribers...)
```
Comment (huanjani) [inline internal/pkg/stream/cloudformation.go:127]: Ah, clever!

**Review by huanjani — APPROVED** [review]
I think it's just a tab thing, but I like how the 'in-progress' timers and 'complete' timers are in different columns, as if they're popping into place when done.

**Comment by efekarakus** [comment]
>  I think it's just a tab thing, but I like how the 'in-progress' timers and 'complete' timers are in different columns, as if they're popping into place when done.

Yeah :'( it's because `tabwriter` doesn't ignore ansi escape sequences so with colors the whole calculation of columns is kinda off. They have a feature request for it https://github.com/golang/go/issues/12073, but I don't think it's gonna be implemented.

I want to send a follow-up PR so that the alignment for the deployment table falls under the first column, right now things are a bit off

File: internal/pkg/deploy/cloudformation/cloudformation.go [hunk 2]
```diff
@@ -307,6 +365,31 @@ func parseStackNameFromARN(stackARN string) string {
 	return strings.Split(stackARN, "/")[1]
 }
 
+// parseEnvStackFromWorkload retrieves the environment stack's name from the workload's stack name.
+// A workload stack name follows the pattern "{env stack name}-{workload name}".
+func parseEnvStackFromWorkload(stackName string) string {
+	parts := strings.Split(stackName, "-")
+	return strings.Join(parts[:len(parts)-1], "-")
+}
```
Comment (iamhopaul123) [inline internal/pkg/deploy/cloudformation/cloudformation.go]: What if the env stack name is `test-env` and workload name is `my-svc`?

File: internal/pkg/stream/cloudformation.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/stream/cloudformation.go:127]: staticcheck yelled at me to make this change :)

**Review by iamhopaul123 — COMMENTED** [review]
Very cool improvement! Would be so much nicer to keep track of the env stack update for users. Just a caveat to bring up.

File: internal/pkg/deploy/cloudformation/cloudformation.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/cloudformation.go]: oh oh good pointttttt 🤔 let me think about this

Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/cloudformation.go]: OK fixed it! I am using the tags in the stack to get the app and env name, let me know your thoughts!

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**